source/jobs/JobsFeature.cpp

```diff
@@ -167,10 +173,19 @@ void JobsFeature::subscribeToUpdateJobExecutionStatusAccepted(string jobId)
         std::bind(
             &JobsFeature::updateJobExecutionStatusAcceptedHandler, this, std::placeholders::_1, std::placeholders::_2),
         std::bind(&JobsFeature::ackSubscribeToUpdateJobExecutionAccepted, this, std::placeholders::_1));
+
+    if (std::future_status::timeout == updateAcceptedPromise.get_future().wait_for(std::chrono::seconds(10)))
```

Comment:
So it seems that the purpose of the promise is to block until the SUBACK message is received and fail if it takes too long but I'm not sure whether `SubscribeToUpdateJobExecutionAccepted` blocks until it receives the SUBACK or not? Because if it blocks, then that promise won't be helpful. Can you refer me to the docs for the behavior of the SubscribeToUpdateJobExecutionAccepted?

I was only able to find the JS docs which does return a promise that resolves when the SUBACK is received but I don't know if the CPP SDK behaves similarly or not.

Reply:
I have no reason to believe that `SubscribeToUpdateJobExecutionAccepted` will actually block until the suback is received. This would be the main motivator for having the sub ack in the first place. The updateAcceptedPromise is set by the `ackSubscribeToUpdateJobExecutionAccepted` function. Therefore, this current code will block on line 177 while waiting for the promise's value to be set.